client/keys.go

```diff
@@ -133,6 +133,25 @@ type WatcherOptions struct {
 	Recursive bool
 }
 
+type CreateOptions struct {
+
+	// TTL defines a period of time after-which the Node should
+	// expire and no longer exist. Values <= 0 are ignored. Given
+	// that the zero-value is ignored, TTL cannot be used to set
+	// a TTL of 0.
+	TTL time.Duration
+
+	// InOrder specifices whether or not the the keys are created
+	// with key names that are in-order.   Generated Key names use
+	// the global etcd index, so the next key can be more than
+	// previous + 1
+	InOrder bool
```

Comment:
This little flag has a major impact on what's going on. I still don't feel good about implementing it this way. Did you think about how this could be implemented if we leave Create alone and add a new CreateInOrder/createInOrderAction?